spectral_cube/spectral_cube.py

```diff
@@ -2519,8 +2519,10 @@ def spectral_interpolate(self, spectral_grid,
             cubedata = self.filled_data[::-1]
```

Comment:
Hmm... no, now that you mention it, I think we could change this to
```
cubedata = self.filled_data
specslice = slice(None) if indiff >=0 else slice(None,None,-1)
```
then later, use
```
                newcube[:,iy,ix] = np.interp(spectral_grid.value, inaxis.value,
                                             cubedata[specslice,iy,ix].value)
```